app/services/sweeper.py

```
@@ -0,0 +1,243 @@
+import asyncio
+import logging
+import random
+import time
+from collections.abc import Callable
+from datetime import datetime, timedelta, timezone
+from typing import AsyncContextManager
```

Ruff (0.14.10)

`typing.AsyncContextManager` is deprecated, use `contextlib.AbstractAsyncContextManager` instead

(UP035)